### ISSUE TITLE
SERVE_PUBLIC Usage

### ISSUE THREAD

**Issue by sabitrakhl** [tfranzel/drf-spectacular#915]
I have used the **SERVE_PUBLIC** settings to hide api endpoints. Now I am not seeing the authorize button. It only shows "No operations defined in spec!" message. Is there any way I can fix this?

**Comment by tfranzel** [comment]
Hi @sabitrakhl,

are you sure nothing else changed on your side? Everything related to this feature has not changed for atleast a year. I cannot see a reason why this should have suddenly stopped working.

Maybe your default authentication method changed and you now also need to set `SERVE_AUTHENTICATION`?

**Comment by sabitrakhl** [comment]
@tfranzel Thank you.

'SERVE_PUBLIC': False,
'SERVE_PERMISSIONS': ['rest_framework.permissions.IsAuthenticated'],
'SERVE_AUTHENTICATION': ['rest_framework.authentication.SessionAuthentication'],

This setting is working now as expected.

**Comment by tfranzel** [comment]
keep `SERVE_PERMISSIONS` on `allowAny` so that the public schema is still requestable for unauthed users

- not authed: public endpoints
- authed: public + private endpoints

**Comment by sayyidfadhlillah** [comment]
Hello @tfranzel and @sabitrakhl,

I still have the same issue even when using your setting:

```
  SPECTACULAR_SETTINGS = {
      'SERVE_INCLUDE_SCHEMA': False,
      'SERVE_PUBLIC': False,
      'SERVE_PERMISSIONS': ['rest_framework.permissions.AllowAny'],
      'SERVE_AUTHENTICATION': ['rest_framework.authentication.SessionAuthentication'],
  }
```

As additional information, this is my DRF Setting:

```
REST_FRAMEWORK = {
    'DEFAULT_SCHEMA_CLASS': 'drf_spectacular.openapi.AutoSchema',
    'DEFAULT_AUTHENTICATION_CLASSES': [
        'rest_framework_simplejwt.authentication.JWTAuthentication',
        'rest_framework.authentication.BasicAuthentication',
        'rest_framework.authentication.SessionAuthentication',
    ],
}
```

am I missing something?

**Comment by tfranzel** [comment]
Hi, what is the issue exactly? No "authorize" button, no operation or no swagger UI showing up at all?

I think we have a issue with only session auth in  `SERVE_AUTHENTICATION`. just leave it on default, then try to authenticate with basic or JWT, which will get properly used by the schema endpoint.

**Comment by sayyidfadhlillah** [comment]
Hi, thank you for the reply.

The issue that I have is that the `Authorize` button is not showing up at all while the rest is working just fine.

I have tried to also leave the `SERVE_AUTHENTICATION` using default but the `Authorize` button still did not showed up.

```
SPECTACULAR_SETTINGS = {
    'SERVE_INCLUDE_SCHEMA': False,
    'SERVE_PUBLIC': False,
    'SERVE_PERMISSIONS': ['rest_framework.permissions.AllowAny'],
}
```

Thus, I am still unsure what I did wrong.

**Comment by tfranzel** [comment]
> Currently, there are no console warning exist.

not the brower console. the server process console. But since you have `jwtAuth` present, there is likely not that much to find there.

> Yes, the authorize button did not exist if SERVE_PUBLIC = False.

Sry, I misspoke. I meant is the button there for `"SERVE_PUBLIC": True`

are you using the most recent spectacular version? really older versions had trouble there.

**Comment by PedroDaumas** [comment]
Hey, I'm also facing this issue!

The button works just fine if `"SERVE_PUBLIC": True` but once I set it to False the button dissapears

**Comment by PedroDaumas** [comment]
I figured out what was causing this issue, I had 2 routes that I wanted to be accessible by anyone so I removed the authentication class and added permission AllowAny to those routes.

The problem is that when I turn "SERVE_PUBLIC" to False the only 2 routes available don't have any authentication, so the swagger page removes the Authorize button. I was able to fix this behavior by keeping the permission AllowAny and having my normal authentication method on my routes.
